src/search/utils/logging.h

```diff
@@ -4,36 +4,51 @@
 #include "system.h"
 #include "timer.h"
 
+#include <memory>
 #include <ostream>
 #include <string>
 #include <vector>
 
 namespace options {
 class OptionParser;
+class Options;
 }
 
 namespace utils {
+// See add_log_options_to_parser for documentation.
+enum class Verbosity {
+    SILENT,
+    NORMAL,
+    VERBOSE,
+    DEBUG
+};
+
 /*
-  Simple logger that prepends time and peak memory info to messages.
-  Logs are written to stdout.
+  Simple line-based logger that prepends time and peak memory info to each line
+  of output. Lines should be eventually terminated by endl. Logs are written to
+  stdout.
 
-  Usage:
-        utils::g_log << "States: " << num_states << endl;
+  Internal class encapsulated by LogProxy.
 */
 class Log {
-private:
-    bool line_has_started = false;
+    std::ostream &stream;
```

Comment:
Looks a bit strange that we store a reference to a stream, but hard code it to std::cout. I know we want to have configurable streams in the future, but who knows when we do this and adding then the stream options would not take a lot of effort.
Thus, I am uncertain, if we should simplify the code to just use std::cout as before.